manager/controllers/default/security/login.class.php

```diff
@@ -460,7 +556,95 @@ public function handleForgotLogin() {
                 $this->setPlaceholder('success_message', $this->modx->lexicon('login_password_reset_act_sent'));
             }
         } else {
-            $this->setPlaceholder('success_message',$this->modx->lexicon('login_user_err_nf_email'));
+            $this->setPlaceholder('success_message', $this->modx->lexicon('login_user_err_nf_email'));
+        }
+    }
+
+
+    /**
+     * Sets an activation/magic login hash for a user.
+     *
+     * @param $hash
+     * @param $user
+     */
+    private function setActivationHash($hash, $user) {
+
+        /** @var modRegistry $registry */
+        $registry = $this->modx->getService('registry', 'registry.modRegistry');
+        /** @var modRegister $register */
+        $register = $registry->getRegister('user', 'registry.modDbRegister');
+        $register->connect();
+        $register->subscribe('/pwd/change/');
```

Comment:
If I'm following the context of this method propertly, `setActivationHash` is the method that creates the one-time login hash. That's not the same action as changing a password (tho I do see that's the way you've implemented this feature). Perhaps a different registry instance/path here would be more appropriate for that.

Reply:
Done.